components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -703,6 +704,120 @@ public void addSubscriber(Subscriber subscriber, String groupingId) throws APIMa
+            rs = ps.executeQuery();
+            if (rs.next()) {
+                MonetizationUsagePublishInfo monetizationUsagePublishInfo = new MonetizationUsagePublishInfo();
+                monetizationUsagePublishInfo.setId(rs.getString("ID"));
+                monetizationUsagePublishInfo.setState(rs.getString("STATE"));
+                monetizationUsagePublishInfo.setStatus(rs.getString("STATUS"));
+                monetizationUsagePublishInfo.setStartedTime(rs.getLong("STARTED_TIME"));
+                monetizationUsagePublishInfo.setLastPublishTime(rs.getLong("LAST_PUBLISHED_TIME"));
+                return monetizationUsagePublishInfo;
+            }
+        } catch (SQLException e) {
+            handleException("Error while retrieving Monetization Usage Publish Info: ", e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(ps, conn, rs);
+        }
+        return null;
+    }
+
+    /**
+     * Add info about monetization usage publish job
+     *
+     * @throws APIManagementException
+     */
+    public void addMonetizationUsagePublishInfo(MonetizationUsagePublishInfo monetizationUsagePublishInfo)
+            throws APIManagementException {
+
+        Connection conn = null;
+        ResultSet rs = null;
+        PreparedStatement ps = null;
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            conn.setAutoCommit(false);
+
+            String query = SQLConstants.ADD_MONETIZATION_USAGE_PUBLISH_INFO;
+            ps = conn.prepareStatement(query);
+
+            ps.setString(1, monetizationUsagePublishInfo.getId());
+            ps.setString(2, monetizationUsagePublishInfo.getState());
+            ps.setString(3, monetizationUsagePublishInfo.getStatus());
+            ps.setString(4, Long.toString(monetizationUsagePublishInfo.getStartedTime()));
+            ps.setString(5, Long.toString(monetizationUsagePublishInfo.getLastPublishTime()));
+            ps.executeUpdate();
+
+            conn.commit();
+        } catch (SQLException e) {
+            if (conn != null) {
+                try {
+                    conn.rollback();
+                } catch (SQLException ex) {
+                    log.error("Error while rolling back the failed operation", ex);
+                }
+            }
+            handleException("Error while adding monetization usage publish Info: ", e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(ps, conn, rs);
+        }
+    }
+
+    /**
+     * Updates info about monetization usage publish job
+     *
+     * @throws APIManagementException
+     */
+    public void updateUsagePublishInfo(MonetizationUsagePublishInfo monetizationUsagePublishInfo)
+                throws APIManagementException {
+
+        Connection conn = null;
+        ResultSet rs = null;
+        PreparedStatement ps = null;
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            conn.setAutoCommit(false);
+
```

Comment:
remove extra space

Reply:
Fixed in 0cb422c3d49f3c8e3e3d4e02e156665fd004836b